File: mybatis-plus-join-test/test-join/src/test/java/com/github/yulichang/test/join/apt/unit/FieldNameTest.java
package com.github.yulichang.test.join.apt.unit;

import com.github.yulichang.extension.apt.toolkit.AptWrappers;
import com.github.yulichang.test.join.dto.AreaDTO;
import com.github.yulichang.test.join.entity.AreaDO;
import com.github.yulichang.test.join.entity.apt.AreaDOCol;
import com.github.yulichang.test.join.entity.apt.UserDtoCol;
import com.github.yulichang.test.join.mapper.AreaMapper;
import com.github.yulichang.test.util.Reset;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.util.List;

@SpringBootTest
public class FieldNameTest {

    @Autowired
    private AreaMapper areaMapper;

    @BeforeEach
    void setUp() {
        Reset.reset();
    }


    @Test
    void testFieldName() {
        AreaDOCol ar = AreaDOCol.build();
        UserDtoCol ud = UserDtoCol.build();
        List<AreaDO> list = areaMapper.selectJoinList(AreaDO.class, AptWrappers.query(ar)
                .select(ar.Postcode)
                .leftJoin(ud, ud.id, ar.id));

        assert list.get(0).getPostcode() != null;
    }

    @Test
    void testFieldName1() {
        AreaDOCol ar = AreaDOCol.build();
        UserDtoCol ud = UserDtoCol.build();
        List<AreaDTO> list = areaMapper.selectJoinList(AreaDTO.class, AptWrappers.query(ar)
                .selectAs(ar.Postcode, AreaDTO::getPostcode)
                .leftJoin(ud, ud.id, ar.id));

        assert list.get(0).getPostcode() != null;
    }
}

File: mybatis-plus-join-test/test-kotlin/src/test/java/com/github/yulichang/test/kt/m/FieldNameTest.kt
package com.github.yulichang.test.kt.m

import com.github.yulichang.extension.kt.toolkit.KtWrappers
import com.github.yulichang.test.kt.dto.AreaDTO
import com.github.yulichang.test.kt.entity.AreaDO
import com.github.yulichang.test.kt.entity.UserDto
import com.github.yulichang.test.kt.mapper.AreaMapper
import com.github.yulichang.test.util.Reset
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class FieldNameTest {
    @Autowired
    private val areaMapper: AreaMapper? = null

    @BeforeEach
    fun setUp() {
        Reset.reset()
    }

    @Test
    fun testFieldName() {
        val list = areaMapper?.selectJoinList(
            AreaDO::class.java, KtWrappers.query(AreaDO::class.java)
                .select(AreaDO::Postcode)
                .leftJoin(UserDto::class.java, UserDto::id, AreaDO::id)
        )
        list?.forEach(System.out::println)
        assert(list?.get(0)?.Postcode != null)
    }

    @Test
    fun testFieldName1() {
        val list = areaMapper?.selectJoinList(
            AreaDTO::class.java, KtWrappers.query(AreaDO::class.java)
                .selectAs(AreaDO::Postcode, AreaDTO::postcode)
                .leftJoin(UserDto::class.java, UserDto::id, AreaDO::id)
        )
        list?.forEach(System.out::println)
        assert(list?.get(0)?.postcode != null)
    }
}
